### PR TITLE
gha: add kubernetes tests workflow for ppc64le

### PR DESCRIPTION
**Pull request by Amulyam24**
This PR adds workflow for running kubernetes test suite on ppc64le. It uses scripts to create and delete the cluster using kubeadm as none of the current cluster creation tools are supported on Power.

Fixes: #7950

### CODE REVIEW

**Comment by Amulyam24** [comment]
Sample run from my fork - https://github.com/Amulyam24/kata-containers/actions/runs/7708735669/job/21010144191

File: .github/workflows/run-k8s-tests-on-ppc64le.yaml
Comment (stevenhorsman) [inline .github/workflows/run-k8s-tests-on-ppc64le.yaml:37]: Was this just hard-coded for testing and should be switched back now?
```suggestion
      DOCKER_REPO: ${{ inputs.repo }}
```

Comment (Amulyam24) [reply .github/workflows/run-k8s-tests-on-ppc64le.yaml]: That's right, I changed it while testing and missed correcting it. Rectified it.

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks

**Review by zvonkok — APPROVED** [review]
LGTM, power -- always reminds me on my work on JS20 and JS21 blades :)

**Comment by Amulyam24** [comment]
some of the required checks are failing with errors not related to this PR, should we wait until they pass or merge this PR?

**Comment by stevenhorsman** [comment]
> some of the required checks are failing with errors not related to this PR, should we wait until they pass or merge this PR?

There is a required test failing which we'll need to pass first, so I'll re-run them, but if it's just non-required we can merge without them

**Comment by Amulyam24** [comment]
Thanks @stevenhorsman @GabyCT ! Merging this PR.

**Pull request merged**
